kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -53,12 +53,26 @@
 class KEVMSemantics(DefaultSemantics):
     auto_abstract_gas: bool
     allow_symbolic_program: bool
-    _cached_subst: Subst | None
+    _custom_step_definitions: tuple[
+        tuple[KSequence, Callable[[Subst, CTerm, CTermSymbolic], KCFGExtendResult | None]], ...
```

Comment:
Consider extracting a class with an attribute (`KSequence`) and a method (`Callable[....]`).

Reply:
Hmm, should I also encapsulate the execution inside of the class? For example, adding a method like 
```py
    def try_execute(self, cterm: CTerm, cterm_symbolic: CTermSymbolic) -> KCFGExtendResult | None:
        subst = self.pattern.match(cterm.cell('K_CELL'))
        if subst is not None:
            return self.exec_fn(subst, cterm, cterm_symbolic)
        return None
```

Reply:
Sounds reasonable to me.

Reply:
I implemented the class. Not sure if I should instantiate the `CustomStep` elements in the `__init__` itself or somewhere else.